diff --git a/node_modules/@azure/core-http/dist-esm/src/util/utils.js b/node_modules/@azure/core-http/dist-esm/src/util/utils.js
index 407e56c..34b73f8 100644
--- a/node_modules/@azure/core-http/dist-esm/src/util/utils.js
+++ b/node_modules/@azure/core-http/dist-esm/src/util/utils.js
@@ -7,10 +7,14 @@ const validUuidRegex = /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F
 /**
  * A constant that indicates whether the environment is node.js or browser based.
  */
+// KLUDGE: @azure/storage-blob uses isNode variable exported from @azure/core-http to
+// determine how it should process data. However, in the renderer process, isNode is
+// set to be true, which causes @azure/storage-blob fails to process data. Thus we need
+// to patch isNode to be false in the renderer process.
+// github issue: https://github.com/Azure/azure-sdk-for-js/issues/21110
 export const isNode = typeof process !== "undefined" &&
-    !!process.version &&
-    !!process.versions &&
-    !!process.versions.node;
+    !!process.env &&
+    !process.env.RENDERER
 /**
  * Checks if a parsed URL is HTTPS
  *
diff --git a/node_modules/@azure/core-http/dist/index.js b/node_modules/@azure/core-http/dist/index.js
index 682b20d..d43a572 100644
--- a/node_modules/@azure/core-http/dist/index.js
+++ b/node_modules/@azure/core-http/dist/index.js
@@ -318,9 +318,9 @@ const validUuidRegex = /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F
  * A constant that indicates whether the environment is node.js or browser based.
  */
 const isNode = typeof process !== "undefined" &&
-    !!process.version &&
-    !!process.versions &&
-    !!process.versions.node;
+    !!process.env &&
+    !process.env.RENDERER;
+
 /**
  * Encodes an URI.
  *
